### PR TITLE
Account for generics when suggesting bound

### LINKED ISSUE #81175: GAT's + const generics error yields Invalid suggestion
The following code errors, and the error contains a suggestion how to fix the error:
```Rust
#![feature(generic_associated_types)]
#![feature(const_generics)]
#![feature(const_evaluatable_checked)]

trait Funk<const P: usize> {
    type Item<const N: usize>: Funk<{core::mem::size_of::<Self>()}>;
}
```

The error message states:
```
error[E0277]: the size for values of type `Self` cannot be known at compilation time
   --> src/lib.rs:6:59
    |
6   |     type Item<const N: usize>: Funk<{core::mem::size_of::<Self>()}>;
    |                                                           ^^^^ doesn't have a size known at compile-time
    |
help: consider further restricting `Self`
    |
5   | trait Funk: Sized<const P: usize> {
    |           ^^^^^^^

error: aborting due to previous error; 3 warnings emitted

For more information about this error, try `rustc --explain E0277`.
error: could not compile `playground`

```

Obviously, writing `trait Funk: Sized<const P: usize> {` is invalid syntax.

### PR DESCRIPTION
**Pull request by estebank**
Fix #81175.

### CODE REVIEW

File: src/test/ui/bound-suggestions.rs [hunk 1]
```diff
@@ -40,4 +40,9 @@ fn test_many_bounds_where<X>(x: X) where X: Sized, X: Sized {
     //~^ ERROR doesn't implement
 }
 
-pub fn main() { }
+trait Foo<T> {
```
Comment (oli-obk) [inline src/test/ui/bound-suggestions.rs:43]: Please also add tests for the other cases (already existing where bounds and already existing super traits). I guess technically even one with both already existing :laughing:

Comment (estebank) [reply src/test/ui/bound-suggestions.rs:43]: Done.

**Pull request merged**